app/views/account/weeklyDetails.scala.html

```diff
@@ -26,16 +26,18 @@ <h3 class="mma-section__header">
                 </h3>
                 @views.html.account.fragments.yourDetails(
                     subscriptionName = subscription.name.get,
-                    plan = subscription.latestPlan,
-                    maybeStartDate = Some(subscription.firstPaymentDate).filter(d => d.isAfter(LocalDate.now.minusMonths(1))),
-                    maybeContact = Some(contact))()
+                    plan = subscription.currentOrExpiredPlan,
+                    maybeStartDate = Some(subscription.currentOrExpiredPlan.start),
+                    maybeContact = Some(contact),
+                    maybeFuturePlan = if (subscription.latestPlan.start.isAfter(now)) Some(subscription.latestPlan) else None
```

Comment:
could have written as Some(subscription.latestPlan).filter(_.start.isAfter(now)) however I'm not sure which is better